pallets/pools/src/lib.rs

```diff
@@ -447,6 +450,9 @@ pub mod pallet {
 		NoSolutionAvailable,
 		/// One of the runtime-level pool parameter bounds was violated
 		PoolParameterBoundViolated,
+		/// Indicates that a pool base currency that is NOT whitelisted was used
+		/// for creating a pool
+		NotAPoolBaseCurrency,
```

Comment:
I'm nitpicking, but I don't think "pool base currency" is a thing, it's just the pool currency (there are no other types of pool currencies).
```suggestion
		NotAValidPoolCurrency,
```

Reply:
Done

Reply:
"InvalidCurrency"

Reply:
I like that.